pkg/cluster_upgrader/cluster_upgrader.go

```diff
@@ -699,15 +702,15 @@ type AlertData struct {
 
 // TODO move to https://github.com/openshift/managed-cluster-validating-webhooks
 // validateUpgradeConfig will validate the UpgradeConfig, the desired version should be grater than or equal to the current version
-func ValidateUpgradeConfig(c client.Client, m maintenance.Maintenance, upgradeConfig *upgradev1alpha1.UpgradeConfig, reqLogger logr.Logger) (result bool, err error) {
+func ValidateUpgradeConfig(c client.Client, ms metrics.Metrics, m maintenance.Maintenance, upgradeConfig *upgradev1alpha1.UpgradeConfig, reqLogger logr.Logger) (result bool, err error) {
 
 	log.Info("validating upgradeconfig")
 	clusterVersion := &configv1.ClusterVersion{}
 	err = c.Get(context.TODO(), types.NamespacedName{Name: "version"}, clusterVersion)
 	if err != nil {
 		log.Info("failed to get clusterversion")
 		log.Error(err, "failed to get clusterversion")
-		metrics.UpdateMetricValidationFailed(upgradeConfig.Name)
+		ms.UpdateMetricValidationFailed(upgradeConfig.Name)
```

Comment:
I'd like to avoid calling to metrics within functions if we can. It's harder to maintain (i.e. someone adds new code and doesn't add a call to metrics)
So for `ValidateUpgradeConfig` and `PostUpgradeVerification` can you use the same pattern as for `PostClusterHealthCheck` where we update metrics based on pass/fail of the entire function?